File: section01/gpio/hacky_mqtt_test.py
import mqtt_helper as mh
import time

def my_first_callback(message_type, payload):
    print("Received the type:", message_type)
    print("Received the payload:", payload)

def main():
    print("Hacky MQTT Test")
    mqtt_client = mh.MqttClient()
    mqtt_client.connect(subscription_topic_name="fisherds",
                        publish_topic_name="fisherds",
                        use_off_campus_broker=True)
    mqtt_client.callback = my_first_callback
    
    counter = 0
    while True:
        time.sleep(2.0)
        counter += 1
        mqtt_client.send_message("number", counter)
        


main()

File: section02/MATLAB_MQTT/hackyMqttTest.m
function hackyMqttTest()
clc
fprintf("Hacky MQTT Test in MATLAB\n")

mqClient = mqttclient("tcp://broker.hivemq.com");

% mqClient.Connected
mqClient.subscribe("fisherds", "Callback", @myCallback)

for k = 1:3
    pause(1)
%     message = sprintf("My number is %d.", k);
%     write(mqClient, "fisherds", message);

    messageStruct.type = "chat";
    messageStruct.payload = sprintf("My number is %d.", k);
    message = jsonencode(messageStruct);
    mqClient.write("fisherds", message);
end
pause(1)  % small delay to receive the last message
end

function myCallback(topic, message)
messageStruct = jsondecode(message);
messageType = messageStruct.type
payload = messageStruct.payload

end
